src/kernel/include/thd.h

```diff
@@ -493,34 +532,56 @@ thd_current_pgtbl(struct thread *thd)
 	/* don't use the cached invstk_top here. We need the stack
 	 * pointer of the specified thread. */
 	curr_entry = &thd->invstk[thd->invstk_top];
-	return curr_entry->comp_info.pgtbl;
+	return curr_entry->comp_invstk_info.comp_info.pgtbl;
+}
+
+/*
+ * This function will be called by thd_invstk_push to modify the current top entry of invstk.
+ * This modification will include the modification of the ip, sp, and in_fault flag.
+ */
+static inline void
+thd_invstk_modify_current(struct thread *thd, unsigned long ip, unsigned long sp, unsigned long in_fault, 
+						  struct cos_cpu_local_info *cos_info)
+{
+	struct invstk_entry *    curr_entry;
+	struct comp_info *       curr_comp_info;
+
+	curr_entry = &thd->invstk[curr_invstk_top(cos_info)];
+	curr_comp_info = &curr_entry->comp_invstk_info.comp_info;
+	curr_entry->ip = ip;
+	curr_entry->sp = sp;
+
+	curr_comp_info->captbl = (struct captbl*)(((unsigned long)curr_comp_info->captbl & (~1u)) | in_fault);
 }
 
+/*
+ * The push operation of invstk has been divided into two steps. First, modify the top of the invstk
+ * by calling the thd_invstk_modify_current fucntion. Then push a new invstk_entry into the invstk.
```

Comment:
nitpicks: 
1. fucntion vs function typo.
2. Do you really need that comment here? Or I think the code is readable to understand that it first modifies the top and then copies the new entry.